File: frontend/src/models/AuthModels.ts
/**
 * Authentication models
 * TypeScript interfaces for authentication-related data
 */

export interface User {
  id: number;
  email: string;
  full_name: string | null;
  is_active: boolean;
  is_admin?: boolean;
  email_verified?: boolean;
  created_at: string;
}

export interface RegisterRequest {
  email: string;
  password: string;
  full_name?: string;
}

export interface LoginRequest {
  username: string; // OAuth2 uses 'username' field for email
  password: string;
}

export interface TokenResponse {
  access_token: string;
  token_type: string;
  user: User;
}

export interface AuthState {
  user: User | null;
  token: string | null;
  isAuthenticated: boolean;
}

File: ios-client/OutfitSuggestor/Models/AuthModels.swift
//
//  AuthModels.swift
//  OutfitSuggestor
//
//  Models for authentication (matches backend API)
//

import Foundation

// MARK: - User (from GET /api/auth/me and Token.user)
struct User: Codable, Identifiable {
    let id: Int
    let email: String
    let full_name: String?
    let is_active: Bool
    let is_admin: Bool?
    let email_verified: Bool?
    let created_at: String
}

// MARK: - Token (from login/register)
struct Token: Codable {
    let access_token: String
    let token_type: String
    let user: User
}

// MARK: - Register request
struct RegisterRequest: Codable {
    let email: String
    let password: String
    let full_name: String?
}

// MARK: - Change password request
struct ChangePasswordRequest: Codable {
    let current_password: String
    let new_password: String
}

// MARK: - API error
struct AuthAPIError: Codable {
    let detail: String
}
